department-of-veterans-affairs/va.gov-cms
# VACMS-5234: Execute all steps of multi-step tasks and return errors appropriately.

## Description

See #5234.  This probably needs to be rewritten further, but there's an open issue for refactoring a lot of this logic.

## Testing done
Ran periodic and daily tasks and verified that exit codes of the command as a whole matched what I saw from stdout/stderr.

## Screenshots
<img width="1178" alt="Screen Shot 2021-05-18 at 11 52 08 AM" src="https://user-images.githubusercontent.com/1318579/118684354-25314a80-b7d0-11eb-8f88-493f8a16679e.png">

## QA steps

As user _uid_ with _user_role_
1. Run `[lando] task --taskfile=tasks-periodic.yml periodic` and confirm that results and exit code match acceptance criteria:
    - [ ] Task <sub>_n_ + 1</sub> from a list of periodic tasks will executed whether Task <sub>_n_</sub> succeeds or fails.
    - [ ] Exit code for the command is 0 iff all tasks pass, otherwise 1.
2. Run `[lando] task --taskfile=tasks-periodic.yml daily` and confirm that results and exit code match acceptance criteria:
    - [ ] Task <sub>_n_ + 1</sub> from a list of periodic tasks will executed whether Task <sub>_n_</sub> succeeds or fails.
    - [ ] Exit code for the command is 0 iff all tasks pass, otherwise 1.

### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
